### PR TITLE
In app notifications

### PR DESCRIPTION
**Pull request by ChrisCruze**
Implemented In App Notifications after user Pushes on Goal including modal popup congratulating user

### CODE REVIEW

**Comment by vivaromarf** [comment]
I ran it and it didn't seem to work right, the confetti did launch faster though.

**Comment by vivaromarf** [comment]
Well it didn't work at all I didn't see a badge or notification in the ios simulator.

**Comment by ChrisCruze** [comment]
> Well it didn't work at all I didn't see a badge or notification in the ios simulator.

Make sure to run the following command: 

`expo install`

**Comment by vivaromarf** [comment]
Screen Shot 2020-08-16 at 9 57 50 PM

Got it to work thanks @ChrisCruze  Looks amazing so does the gear symbol and new pushable button @anandangalig

File: Components/Atoms/NotificationsState.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
+import React, { useState } from "react";
```
Comment (anandangalig) [inline Components/Atoms/NotificationsState.js:1]: do you need to import React and useState here? I don't see it being used here

File: Components/Atoms/NotificationsState.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+  return time_stamp_count == 1;
+};
+
+const roundNumber = (x, divisor) => {
+  return Math.ceil(x / divisor) * divisor;
+};
+
+const determineIfStreak = ({ streakCount }) => {
+  const is_divisible_by_5 = roundNumber(streakCount, 5) == streakCount;
+  return streakCount == 2 || is_divisible_by_5 & (streakCount > 2);
```
Comment (anandangalig) [inline Components/Atoms/NotificationsState.js:14]: what does **&** do here? Is it same as &&?

File: Components/Atoms/NotificationsState.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+const streakText = ({ streakCount, cadence }) => {
+  return `Yo Ho Ho - you're on ${streakCount} ${cadence} streak`;
+};
+
+const tenThresholdText = ({ time_stamp_count }) => {
+  return `Ahoy - ye just reached ${time_stamp_count} pushes`;
+};
+
+const maxPushesText = ({ time_stamp_count }) => {
+  return `Avast Ye - ye just a new personal record of ${time_stamp_count} pushes for a single goal `;
```
Comment (anandangalig) [inline Components/Atoms/NotificationsState.js:40]: ye just a new personal  - ye just **achieved** a new personal ?

**Comment by anandangalig** [comment]
Here are my thoughts:

1. Every single push has a popup, either smaller one up top or big one with a start. I think its too much. These types of things need to be mark a milestone and rendered every once in a while. 

2. 
IMG_3971

3.
IMG_3970

File: Components/Atoms/NotificationsState.js (hunk 1 above)
Comment (ChrisCruze) [reply Components/Atoms/NotificationsState.js:1]: good catch

File: Components/Atoms/NotificationsState.js (hunk 3 above)
Comment (ChrisCruze) [reply Components/Atoms/NotificationsState.js:40]: good catch

**Comment by ChrisCruze** [comment]
> * Every single push has a popup, either smaller one up top or big one with a star. I think its too much. These types of things need to be marking a milestone and rendered every once in a while.
> *

The reason I like the small ones on top because I'd like to know my "score" immediately after I push, so I'd like to keep it unless we can come up with another solution that gives user immediate feedback after they push. I agree that the big one needs to come up less often though.

**Comment by ChrisCruze** [comment]
I believe I addressed the majority of concerns here. Modal now only pops up when daily, weekly, monthly goals are complete. Also added a timer to it to make it better experience. In App Notification still pops up at the top though just so user can get immediate feedback

**Pull request merged**